core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -367,6 +369,57 @@ else if ( store instanceof Group )
         }
     }
 
+    public synchronized void rescheduleDisableTimeout( final StoreKey key )
+            throws IndySchedulerException
+    {
+        if ( !schedulerConfig.isEnabled() )
+        {
+            logger.debug( "Scheduler disabled." );
+            return;
+        }
+
+        ArtifactStore store = null;
+        try
+        {
+            store = dataManager.getArtifactStore( key );
+        }
+        catch ( final IndyDataException e )
+        {
+            logger.error( String.format( "Failed to retrieve store for: %s. Reason: %s", key, e.getMessage() ), e );
+        }
+
+        if ( store == null )
+        {
+            return;
+        }
+
+        int timeout = store.getDisableTimeout();
+        if ( timeout == TIMEOUT_USE_DEFAULT )
+        {
+            // case TIMEOUT_USE_DEFAULT: will use default timeout configuration
+            timeout = config.getStoreDisableTimeoutSeconds();
+        }
+
+        // No need to cancel as the job will be cancelled immediately after the re-enable in StoreEnablementManager
+//        final Set<ScheduleKey> canceled =
+//                cancelAllBefore( new StoreKeyMatcher( store.getKey(), DISABLE_TIMEOUT ),
+//                                 timeout );
+//        logger.info( "Cancel disable timeout for stores:{}", canceled );
+
+        if ( timeout > TIMEOUT_USE_DEFAULT && store.isDisabled() )
+        {
+            final StoreKey sk = store.getKey();
+            logger.debug( "Set/Reschedule disable timeout for store:{}", sk );
+            scheduleForStore( sk, DISABLE_TIMEOUT, DISABLE_TIMEOUT, sk, timeout );
+        }
+        else if ( timeout <= TIMEOUT_NEVER_DISABLE )
+        {
+            logger.debug( "Disable-timeout set to {}, will never disable the repo", timeout );
+            store.setDisabled( false );
```

Comment:
Of course, it might just be simplest if we check for `store.getDisableTimeout() == TIMEOUT_NEVER_DISABLE` in the observer call, and re-enable the store there. It's a bit clumsy, but a lot simpler than the stuff I mentioned above...